### ISSUE TITLE
Alias identified as unused and potentially stripped from VALUES derived table

### ISSUE THREAD

**Issue by zalky** [sqlfluff/sqlfluff#5600]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I ran sql fluff on the following query:

```sql
SELECT *
FROM (
    VALUES
    (1, 2),
    (3, 4)
) AS t (x, y)
```
And it identified `t` as never used, but postgres requires that derived tables have an alias.

### Expected Behaviour

The alias should not be removed from VALUES derived tables.

### Observed Behaviour

The VALUES derived table alias is identified as unused (and potentially removed). I can confirm that UNION derived table aliases are _not_ removed (fixed by #3547 )

### How to reproduce

This is a variation of the following two issues:

- https://github.com/sqlfluff/sqlfluff/issues/3545
- https://github.com/sqlfluff/sqlfluff/issues/3547

```console
$ cat sqlfluff.config
[sqlfluff]
dialect = postgres
templater = raw
$ cat sqlfluff.sql
SELECT *
FROM (
    VALUES
    (1, 2),
    (3, 4)
) AS t (x, y)
$ sqlfluff fix sqlfluff.sql --config sqlfluff.config 
==== finding fixable violations ====
== [sqlfluff.sql] FAIL
L:   6 | P:   6 | AL05 | Alias 't' is never used in SELECT statement.
                       | [aliasing.unused]
==== fixing violations ====
1 fixable linting violations found
Are you sure you wish to attempt to fix these? [Y/n] 
```

### Dialect

postgres

### Version

sqlfluff, version 2.3.4
Python 3.10.9

### Configuration

[sqlfluff]
dialect = postgres
templater = raw

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by zalky** [comment]
Solution along these lines?

https://github.com/sqlfluff/sqlfluff/pull/3548/files
